src/lib/Form/Type/Language/ConfiguredLanguagesChoiceType.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\Type\Language;
+
+use EzSystems\EzPlatformAdminUi\Form\Type\ChoiceList\Loader\ConfiguredLanguagesChoiceLoader;
+
+/**
+ * Form Type allowing to select from all configured (also not enabled) Languages.
+ */
+class ConfiguredLanguagesChoiceType extends LanguageChoiceTypeAbstract
```

Comment:
The implementation of `ConfiguredLanguagesChoiceType` and `LanguageChoiceType` is almost identical, so it feels redundant. How does it happen that for `ConfiguredLanguagesChoiceType` the `ConfiguredLanguagesChoiceLoader` is injected instead? Based on the usage I don't see it (expected at least `choice_loader` option in `\EzSystems\EzPlatformAdminUi\Form\Type\Search\SearchType::buildForm`).

Reply:
`ConfiguredLanguagesChoiceType` and `LanguageChoiceType` have injected different ChoiceLoaders, with a little different strategy. Above classes are autowiring by typehint.

Reply:
> Above classes are autowiring by typehint.

Where exactly does the auto-wiring for that occur? The proper solution in this case is not to auto-wire, but inject directly the specific implementation (so single constructor in abstract class only and direct injection of `LanguageChoiceLoader` and `ConfiguredLanguagesChoiceLoader`). Still, can't find a place where it should happen. Form type is used as `<ClassName>::class` so seems like constructor is unused.